Modules/Core/Common/src/itkSpatialOrientationAdapter.cxx

```diff
@@ -32,47 +32,70 @@ namespace itk
 SpatialOrientationAdapter::OrientationType
 SpatialOrientationAdapter::FromDirectionCosines(const DirectionType & Dir)
 {
```

Comment:
Commit message: The old algorithm could generate an invalid combinations of axes.